synapse/federation/federation_client.py

```diff
@@ -686,6 +686,12 @@ async def _try_destination_list(
         if failover_errcodes is None:
             failover_errcodes = ()
 
+        if not destinations:
+            # Give a bit of a clearer message if no servers were specified at all.
+            raise SynapseError(
+                502, f"Failed to {description} via any server: No servers specified."
+            )
```

Comment:
Is there a suitable matrix error code here? M_MISSING_PARAM?

Reply:
(and should this be a 400 rather than a 502?)

Reply:
similar story here — I would be happy using `M_MISSING_PARAM` + 400 or whatever but only if this is the client trying to join a room ID, but actually doing that is a bit more involved as we have to detect that at the REST layer (and I feel less confident about whether I should be breaking local room joins via ID with no server names. I'm not sure they're actually legal, but whether they get used or not is a different story)